File: clients/android-native/app/src/main/java/com/trailnote/android/data/service/TrailNoteServices.kt
package com.trailnote.android.data.service

import com.trailnote.android.core.network.PageEnvelope

interface AuthService {
    fun login(account: String, password: String): TrailAuthSession
}

interface RouteService {
    fun featuredRoutes(): List<TrailRouteSummary>

    fun routes(page: Int = 1, pageSize: Int = 10): PageEnvelope<TrailRouteSummary>

    fun search(keyword: String): TrailSearchResult

    fun routeDetail(routeId: Long): TrailRouteDetail

    fun toggleRouteFavorite(routeId: Long): TrailFavoriteToggleResult

    fun routeComments(routeId: Long, page: Int = 1, pageSize: Int = 20): PageEnvelope<TrailRouteComment>

    fun addRouteComment(routeId: Long, content: String): TrailCommentCreateResult
}

interface ProfileService {
    fun myProfile(): TrailProfile

    fun myFavorites(page: Int = 1, pageSize: Int = 10): PageEnvelope<TrailRouteSummary>
}

interface AuthorService {
    fun authorProfile(authorId: Long): TrailAuthorProfile

    fun toggleAuthorFollow(authorId: Long): TrailFollowToggleResult
}

File: clients/ios-native/TrailNoteApp/Core/Services/TrailNoteServices.swift
import Foundation

protocol TrailAuthServicing {
    func login(account: String, password: String) async throws -> TrailAuthSessionDTO
}

protocol TrailRouteServicing {
    func featuredRoutes() async throws -> [TrailRouteSummaryDTO]
    func routes(page: Int, pageSize: Int) async throws -> PageEnvelope<TrailRouteSummaryDTO>
    func search(keyword: String) async throws -> TrailSearchResultDTO
    func routeDetail(routeId: Int) async throws -> TrailRouteDetailDTO
    func toggleRouteFavorite(routeId: Int) async throws -> TrailFavoriteToggleResultDTO
    func routeComments(routeId: Int, page: Int, pageSize: Int) async throws -> PageEnvelope<TrailRouteCommentDTO>
    func addRouteComment(routeId: Int, content: String) async throws -> TrailCommentCreateResultDTO
}

protocol TrailProfileServicing {
    func myProfile() async throws -> TrailProfileDTO
    func myFavorites(page: Int, pageSize: Int) async throws -> PageEnvelope<TrailRouteSummaryDTO>
}

protocol TrailAuthorServicing {
    func authorProfile(authorId: Int) async throws -> TrailAuthorProfileDTO
    func toggleAuthorFollow(authorId: Int) async throws -> TrailFollowToggleResultDTO
}
